lib/src/test/java/io/ably/lib/test/realtime/RealtimeChannelTest.java

```diff
@@ -943,4 +943,78 @@ public void detach_success_callback_detached() throws AblyException {
 				ably.close();
 		}
 	}
+
+	/**
+	 * <p>
+	 * Validate publish will result in an error, when the channel is
+	 * in or moves to the FAILED state before the operation succeeds
+	 * </p>
+	 * <p>
+	 * Spec: RTL6c3
+	 * </p>
+	 *
+	 * @throws AblyException
+	 */
+	@Test
```

Comment:
The test case verifies that,
  In case of fail prune channel (in our case invalid test token) implicit attach call made by `publish(...)` with fail in order `attaching` --> `failed` with ErrorInfo. 

Please correct me if I am wrong. Publish won't throw an exception, but will just fail with `ErrorInfo`